src/main/java/org/lateralgm/joshedit/JoshText.java

```diff
@@ -465,6 +459,16 @@ public void caretUpdate(CaretEvent e) {
   // == Settings functions =========================================================================
   // ===============================================================================================
 
+  @Override
+  public void setFont(Font font) {
+    super.setFont(font);
+    FontMetrics fm = getFontMetrics(font);
+    lineAscent = fm.getAscent();
+    lineHeight = fm.getHeight();
+    lineLeading = fm.getLeading();
+    monoAdvance = fm.getWidths()['M'];
+  }
```

Comment:
Not sure if a repaint or code size might be a good idea here. Though LGM is not yet adding the ability to change it on already opened code editors.